src/MongoDB.Analyzer/Core/Utilities/SymbolExtensions.cs

```diff
@@ -20,12 +20,29 @@ internal static class SymbolExtensions
     private const string NamespaceMongoDBDriver = "MongoDB.Driver";
     private const string NamespaceMongoDBLinq = "MongoDB.Driver.Linq";
     private const string NamespaceSystemLinq = "System.Linq";
+    private const string AssemblyMongoDBBson = "MongoDB.Bson";
+    private const string NamespaceMongoDBBson = "MongoDB.Bson.Serialization.Attributes";
 
     private static readonly HashSet<string> s_supportedCollections = new()
     {
         "System.Collections.Generic.List<T>",
         "System.Collections.Generic.IList<T>",
-        "System.Collections.Generic.IEnumerable<T>"
+        "System.Collections.Generic.IEnumerable<T>",
+        "System.Collections.Generic.Dictionary<TKey, TValue>",
+        "System.Collections.Generic.IDictionary<TKey, TValue>"
+    };
```

Comment:
If we are adding more collections support, we need to test that extensively in LinqCollections and `BuildersCollections`.